packages/common/src/ContractInteractor.ts

```diff
@@ -1174,3 +1174,22 @@ export function getRawTxOptions (chainId: number, networkId: number, chain?: str
       }, 'london')
   }
 }
+
+export async function getChainId (provider: Web3ProviderBaseInterface): Promise<number> {
```

Comment:
This particular one is not necessary. This is done to avoid calling `new Web3(provider).eth.getChainId()`, which is not costly